src/sqlite/framekm.ts

```diff
@@ -0,0 +1,196 @@
+import { getConfig } from 'util/motionModel/config';
+import { db, getAsync, runAsync } from './index';
+import { FramesMetadata } from 'types/motionModel';
+
+export const isFrameKmComplete = async (): Promise<boolean> => {
+  try {
+    const row: any = await getAsync(
+      db,
+      'SELECT COUNT(*) AS count FROM framekm;',
+    );
+    const count = row[0].count;
+    return count * getConfig().DX > 1000; // Check if we have at least KM of data
+  } catch (error) {
+    console.error('Error checking if framekm is complete:', error);
+    throw error;
+  }
+};
+
+export const getFrameKmMetadata = async (): Promise<FramesMetadata[]> => {
+  try {
+    const rows: any = await getAsync(
+      db,
+      'SELECT * FROM framekm ORDER BY id;',
+    );
+    return rows;
+  } catch (error) {
+    console.error('Error fetching framekm metadata:', error);
+    return [];
+  }
+}
+
+export const getLastTimestamp = async (): Promise<number> => {
+  try {
+    let row: any = await getAsync(
+      db,
+      'SELECT ts FROM framekm ORDER BY id DESC LIMIT 1;',
+    );
+    if (row.length) {
+      return row[0].ts;
+    }
+
+    // If not found, try to get the last timestamp from the prev_framekm table
+    row = await getAsync(
+      db,
+      'SELECT ts FROM prev_framekm ORDER BY id DESC LIMIT 1;',
+    );
+    if (row.length) {
+      return row[0].ts;
+    }
+
+    // If no timestamps are found in either table, return the current time
+    return Date.now();
+  } catch (error) {
+    console.error('Error fetching last timestamp:', error);
+    return Date.now();
+  }
+};
+
+export const getExistingFramesMetadata = async (limit = 3): Promise<any[]> => {
+  try {
+    let rows: any = await getAsync(
+      db,
+      `SELECT * FROM framekm ORDER BY id DESC LIMIT ${limit};`,
+    );
+
+    if (rows?.length < limit) {
+      // If no records are found in framekm, fetch from prev_framekm
+      const prevRows: any = await getAsync(
+        db,
+        `SELECT * FROM prev_framekm ORDER BY id DESC LIMIT ${limit};`,
+      );
+      rows = rows.concat(prevRows);
+    }
+
+    return rows; // Returns either the last N records from framekm, prev_framekm, or an empty array
+  } catch (error) {
+    console.error('Error fetching frames:', error);
+    return [];
+  }
+};
+
+let metersTrimmed = 0;
+
+export const addFramesToFrameKm = async (rows: FramesMetadata[], tableName = 'framekm'): Promise<void> => {
+  const {
+    isTripTrimmingEnabled,
+    TrimDistance
+  } = getConfig();
+
+  if (isTripTrimmingEnabled && metersTrimmed < TrimDistance) {
```

Comment:
Trip trimming logic lives here for now.
Every time we try to insert new frames into FrameKM table,
if trip trimming is enabled and it was not yet trimmed enough during this session,
it will push those frames into `prev_framekm` table instead of `framekm` table.